shared/surface_flux.F90

```diff
@@ -995,4 +1024,75 @@ subroutine ncar_ocean_fluxes_multilevel (u_del, t, ts, q, qs, zu, zt, zq, avail,
 
 end subroutine ncar_ocean_fluxes_multilevel
 
+subroutine iter_monin_obukhov_ocean (                      &
+           z_atm, u_atm, v_atm, w_atm, thv_atm, q_atm,     &
+           u_surf, v_surf, thv_surf, q_surf0,              &
+           rough_mom, rough_heat, rough_moist,             &
+           cd_m, cd_t, cd_q, u_star, b_star, avail, seawater)
+!~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~!
+!  Right now, it  is only effective when ocean_rough = 'hwrf17', but this
+!  can be expanded if necessarily to incorporate other roughness schemies
+!  contact: Kun.Gao@noaa.gov; Baoqiang.Xiang@noaa.gov
+!~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~!
+
+  real   , intent(in), dimension(:)    ::                  &
+           z_atm, u_atm, v_atm, w_atm, thv_atm, q_atm,     &
+           u_surf, v_surf, thv_surf, q_surf0
+
+  real   , intent(inout), dimension(:) ::                  &
+           rough_mom, rough_heat, rough_moist,             &
+           cd_m, cd_t, cd_q, u_star, b_star
+  logical, intent(in), dimension(:)    ::                  &
+           avail, seawater
+
+  ! local var
+  real, dimension(size(z_atm(:)))      ::                  &
+           flux_q, q_star,                                 &
+           ref_u, ref_v, u10, del_m, del_h, del_q,         &         
+           rough_mom1, rough_heat1, rough_moist1
+  integer i, j
```

Comment:
All of these variables need to be documented in doxygen style because this is a new routine